src/routes/jira/connect/enterprise/app/jira-connect-enterprise-app-put.test.ts

```diff
@@ -0,0 +1,186 @@
+	let jwt: string;
+	const uuid = "4c74e9ce-faf9-489e-821f-e7a8006e473a";
+
+	beforeEach(async () => {
+		installation = await Installation.install({
+			host: jiraHost,
+			sharedSecret: "shared-secret",
+			clientKey: "client-key"
+		});
+
+		app = express();
+		app.use((req: Request, res: Response, next: NextFunction) => {
+			res.locals = { installation };
+			req.log = getLogger("test");
+			req.session = { jiraHost };
+			req.params = { uuid };
+			next();
+		});
+		app.use(RootRouter);
+
+		jwt = encodeSymmetric({
+			qsh: "context-qsh",
+			iss: jiraHost
+		}, installation.sharedSecret);
+	});
+
+	it("should return 200 with success true when correct uuid and installation id are passed", async () => {
+		await GitHubServerApp.create({
+			uuid,
+			appId: 1,
+			gitHubAppName: "my awesome app",
+			gitHubBaseUrl: "http://myinternalinstance.com",
+			gitHubClientId: "lvl.1n23j12389wndd",
+			gitHubClientSecret: "secret",
+			webhookSecret: "anothersecret",
+			privateKey: "privatekey",
+			installationId: installation.id
+		});
+
+		const payload ={
+			gitHubAppName: "my-app",
+			webhookSecret: `encrypted:${uuid}`,
+			appId: "1",
+			gitHubClientId: "Iv1.msdnf2893rwhdbf",
+			gitHubClientSecret: "encrypted:secret",
+			uuid,
+			gitHubBaseUrl: "http://testserver.com",
+			privateKey: "privatekeycontents",
+			jiraHost
+		};
+
+		return supertest(app)
+			.put(`/jira/connect/enterprise/app/${uuid}`)
+			.query({
+				jwt
+			})
+			.send(payload)
+			.expect(200)
+			.then((res) => {
+				expect(res.body.success).toBeTruthy();
+			});
+	});
+
+	it("should return 200 with success true when correct uuid and installation id are passed, with partial data", async () => {
+		await GitHubServerApp.create({
+			uuid,
+			appId: 1,
+			gitHubAppName: "my awesome app",
+			gitHubBaseUrl: "http://myinternalinstance.com",
+			gitHubClientId: "lvl.1n23j12389wndd",
+			gitHubClientSecret: "secret",
+			webhookSecret: "anothersecret",
+			privateKey: "privatekey",
+			installationId: installation.id
+		});
+
+		const payload ={
+			gitHubAppName: "my-app",
+			webhookSecret: `encrypted:${uuid}`,
+			gitHubClientId: "Iv1.msdnf2893rwhdbf",
+			gitHubClientSecret: "encrypted:secret",
+			uuid,
+			jiraHost
+		};
+
+		return supertest(app)
+			.put(`/jira/connect/enterprise/app/${uuid}`)
+			.query({
+				jwt
+			})
+			.send(payload)
+			.expect(200)
+			.then((res) => {
+				expect(res.body.success).toBeTruthy();
+			});
+	});
+
+	it("should return 200 with success false when wrong uuid param is passed", async () => {
+		await GitHubServerApp.create({
+			uuid,
+			appId: 1,
+			gitHubAppName: "my awesome app",
+			gitHubBaseUrl: "http://myinternalinstance.com",
+			gitHubClientId: "lvl.1n23j12389wndd",
+			gitHubClientSecret: "secret",
+			webhookSecret: "anothersecret",
+			privateKey: "privatekey",
+			installationId: installation.id
+		});
+
+		const payload ={
+			gitHubAppName: "my-app",
+			webhookSecret: `encrypted:${uuid}`,
```

Comment:
not a big problem, but at this stage, the "webhookSecret" should be plain text. But technically, this test still works.